nengo_spinnaker/builder/model.py

```diff
@@ -0,0 +1,292 @@
+            sp, _ = tp_sinks.parameters
+
+            # For each sink, if the sink object is the specified object
+            # then add signal to the list.
+            for sink in tp_sinks.sinks:
+                if sink.sink_object is sink_object:
+                    # This is the desired sink object, so remember the
+                    # signal. First construction the reception
+                    # specification.
+                    signals[sink.port].append(
+                        ReceptionSpec(sp, sink.reception_parameters)
+                    )
+
+        return signals
+
+    def get_signals(self):
+        """Extract all the signals from the connection map.
+
+        Yields
+        ------
+        Signal
+            Signal objects derived from the contents of the connection map.
+        """
+        # For each source object and set of sinks yield a new signal
+        for source, port_conns in iteritems(self._connections):
+            # For each connection look at the sinks and the signal parameters
+            for (sig_pars, _), par_sinks in chain(*itervalues(port_conns)):
+                # Create a signal using these parameters
+                yield Signal(
+                    source,
+                    (ps.sink_object for ps in par_sinks),  # Extract the sinks
+                    sig_pars.keyspace,
+                    sig_pars.weight
+                )
+
+
+class OutputPort(enum.Enum):
+    """Indicate the intended transmitting part of an executable."""
+    standard = 0
+    """Standard, value-based, output port."""
+
+
+class InputPort(enum.Enum):
+    """Indicate the intended receiving part of an executable."""
+    standard = 0
+    """Standard, value-based, output port."""
+
+
+class SignalParameters(object):
+    """Basic parameters that can be applied to a signal.
+
+    Attributes
+    ----------
+    latching : bool
+        If False (the default) any receiving filters should reset their inputs
+        every time-step.  If True then receiving filters should hold their
+        inputs until new values are received.
+    weight : int
+        Number of packets expected to be sent every time-step.
+    keyspace : :py:class:`~rig.bitfield.BitField` or None
+        Keyspace which will be used to assign keys to packets.
+    """
+    def __init__(self, latching=False, weight=0, keyspace=None):
+        # Store the parameters
+        self.latching = latching
+        self.weight = weight
+        self.keyspace = keyspace
+
+    def __eq__(self, other):
+        # Equivalent if the latching spec is the same, the weight is the same
+        # and the keyspaces are equivalent.
+        if ((self.latching is other.latching) and
+                (self.weight == other.weight) and
+                (self.keyspace == other.keyspace)):
+            return True
+        else:
+            return False
+
+    def __ne__(self, b):
+        return not self == b
+
+
+class TransmissionParameters(object):
```

Comment:
None of the classes I (by their names) should derive from this actually seem to! Should this class just be deleted or is magic occurring somewhere I missed?